### PR TITLE
[doc/getting_started] Updates and minor corrections to documentation

### PR DESCRIPTION
**Pull request by alees24**
Fixing broken links and minor errors observed whilst getting up to speed. Additional comments re later Ubuntu (20.04) and minimum system requirements.

Signed-off-by: Adrian Lees <a.lees@lowrisc.org>

### CODE REVIEW

File: doc/getting_started/setup_dv.md [hunk 1]
```diff
@@ -66,22 +66,23 @@ There is support for Cadence Xcelium as well, which is being slowly ramped up.
...
 
 The generated initial testbench is not expected to compile and elaborate successfully right away.
-There may be additional fixes required, which can be hopefully be identified easily.
+There may be additional fixes required, which can hopefully be identified easily.
 Once the testbench compiles and elaborates without any errors or warnings, the sanity sequence can be developed further to access a major datapath and test the basic functionality of the DUT.
 
 VCS is used as the default simulator.
 It can be switched to Xcelium by setting `--tool xcelium` on the command line.
-To dump waves from the simulation, pass the `--waves` argument to `dvsim.py`.
+To dump waves from the simulation, pass the `--waves <format>` argument to `dvsim.py`. With Xcelium you should probably use '--waves shm' and with VCS specify '--waves vpd' but both can generate the more portable, but much larger, VCD format by specifying the '--waves vcd' parameter.
```
Comment (timothytrippel) [inline doc/getting_started/setup_dv.md:78]: or `fsdb` for VCS if you using Verdi? @sriyerg

File: doc/getting_started/setup_dv.md [hunk 2]
```diff
@@ -114,5 +115,5 @@ $ util/dvsim/dvsim.py hw/top_earlgrey/dv/top_earlgrey_sim_cfgs.hjson -i smoke --
 We can only run the failed test with `--fixed-seed=1` to reproduce the failure.
 Assume there is a failure in the `uart_smoke` test. Here is the run command to reproduce it.
 ```console
-$ util/dvsim/dvsim.py hw/ip/uart/dv/uart_sim_cfg.hjson -i uart_smoke --fixed-seed=1 [--waves]
+$ util/dvsim/dvsim.py hw/ip/uart/dv/uart_sim_cfg.hjson -i uart_smoke --fixed-seed=1 [--waves vcd]
```
Comment (timothytrippel) [inline doc/getting_started/setup_dv.md:118]: VCD format is huge (orders of magnitude sometimes) compared to others. Do we want this to be what we show in the example over say `fsdb` or `vpd`? @sriyerg can comment more

**Review by timothytrippel — APPROVED** [review]

**Comment by timothytrippel** [comment]
Thanks for the fixes @alees24 !

File: doc/getting_started/setup_dv.md (hunk 2 above)
Comment (alees24) [reply doc/getting_started/setup_dv.md:118]: VCD files are indeed huge, and I'd rather avoid any mention of vcd files. I was only trying to offer a command that works whichever simulator is being used. I could flesh out the description to describe explicitly which simulator and format combinations are available/supported?

File: doc/getting_started/setup_dv.md (hunk 1 above)
Comment (sriyerg) [reply doc/getting_started/setup_dv.md:78]: Perhaps we could just list the supported formats and let the user pick whichever their setup supports (not everyone will have access to a Verdi license (sold separately by Snps). 

```suggestion
To dump waves from the simulation, pass the `--waves <format>` argument to `dvsim.py`. 
- With VCS, `vpd`, and `fsdb` are the supported formats.
- With Xcelium,  `shm`, `fsdb` and `vcd` are the supported formats.
- To find the list of supported formats for each tool, please lookup the `supported_wave_formats` key in the tool specific `dvsim.py` setup at `hw/dv/tools/dvsim/<tool>.hjson`.
```

File: doc/getting_started/setup_dv.md (hunk 2 above)
Comment (sriyerg) [reply doc/getting_started/setup_dv.md:118]: I think we can just leave this as `[--waves <format>]`, if we are already mentioning the list of supported formats above.

**Comment by timothytrippel** [comment]
Looks like you will also need to hit these files with the `util/fix_trailing_whitespace.py` script to get them to pass CI.

File: doc/getting_started/setup_verilator.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ git checkout v$VERILATOR_VERSION
 autoconf
 CC=gcc-11 CXX=g++-11 ./configure --prefix=/tools/verilator/$VERILATOR_VERSION
 CC=gcc-11 CXX=g++-11 make
-CC=gcc-11 CXX=g++-11 make install
+sudo CC=gcc-11 CXX=g++-11 make install
```
Comment (marnovandermaas) [inline doc/getting_started/setup_verilator.md:33]: I would be hesitant to put `sudo` in code that people might copy and paste, I would prefer to add a note in the text below.

File: doc/getting_started/_index.md [hunk 4]
```diff
@@ -193,14 +196,26 @@ You can download and build Verible from scratch as explained on the [Verible Git
 But since this requires the Bazel build system the recommendation is to download and install a pre-built binary as described below.
 
 Go to [this page](https://github.com/google/verible/releases) and download the correct binary archive for your machine.
-The example below is for Ubuntu 18.04:
+
+The example below is for Ubuntu 20.04:
+
+```
+export VERIBLE_VERSION=v0.0-2479-g92928558
```
Comment (mundaym) [inline doc/getting_started/_index.md:203]:
```suggestion
export VERIBLE_VERSION={{< tool_version "verible" >}}
```
I'd prefer to use the generated variable here so that this doesn't get out of date. I've used the current min version (`v0.0-2135-gb534c1fe`) on Ubuntu 20.04.x.

File: doc/getting_started/setup_dv.md [hunk 5]
```diff
@@ -111,8 +111,15 @@ $ git checkout -b <temp_branch> FETCH_HEAD
 ```console
 $ util/dvsim/dvsim.py hw/top_earlgrey/dv/top_earlgrey_sim_cfgs.hjson -i smoke --fixed-seed=1
 ```
-We can only run the failed test with `--fixed-seed=1` to reproduce the failure.
-Assume there is a failure in the `uart_smoke` test. Here is the run command to reproduce it.
+Since the CI runs tests with pseudo-random behaviour driven from 'seed' numbers, to be confident of reproducing the failure we must supply the exact seed that CI used.
+
+Assume there is a failure in the `uart_smoke` test. To reproduce this with the DV simulation environment we use the following command, remembering to replace '<seed>' with the seed number, and to choose an appropriate waveform '<format>':
+
+
```
Comment (mundaym) [inline doc/getting_started/setup_dv.md:118]: Did you mean to put an extra newline here? If not we should stick to one for consistency.

File: hw/dv/dpi/usbdpi/usbdpi.h [hunk 6]
```diff
@@ -93,7 +94,6 @@
 #define HS_NEXTFRAME 14
 
 #define SEND_MAX 32
-#include <stdint.h>
```
Comment (mundaym) [inline hw/dv/dpi/usbdpi/usbdpi.h:96]: This seems like a reasonable change but I think it should be in a separate PR.

File: sw/device/tests/usbdev_test.c [hunk 7]
```diff
@@ -124,10 +124,10 @@ bool test_main(void) {
   base_printf("\r\n");
   for (int i = 0; i < kExpectedUsbCharsRecved; i++) {
     CHECK(buffer[i] == kExpectedUsbRecved[i],
-          "Recieved char #%d mismatched: exp = %x, actual = %x", i,
+          "Received char #%d mismatched: exp = %x, actual = %x", i,
```
Comment (mundaym) [inline sw/device/tests/usbdev_test.c:127]: Same note as above, I think the USB code changes should be put in a separate PR.

**Review by mundaym — COMMENTED** [review]
Thanks Adrian! I've added a couple of minor comments. Once they are addressed I think we can merge this.

File: doc/getting_started/_index.md (hunk 4 above)
Comment (alees24) [reply doc/getting_started/_index.md:203]: I'd guessed (perhaps wrongly?) that the magic 'tool_version' picks up whatever is presently installed on the machine(s) that serve up the web documentation; is that not the case?
My understanding is that CI runs on Ubuntu 18.04, and my concern was that Ubuntu 20.04 would/may require a different version of Verible. Have now altered as requested and trust that version and OS build are orthogonal.

Comment (mundaym) [reply doc/getting_started/_index.md:203]: > I'd guessed (perhaps wrongly?) that the magic 'tool_version' picks up whatever is presently installed on the machine(s) that serve up the web documentation; is that not the case?

I believe the version is taken from this file: `https://github.com/lowRISC/opentitan/blob/master/tool_requirements.py`

> My understanding is that CI runs on Ubuntu 18.04, and my concern was that Ubuntu 20.04 would/may require a different version of Verible. Have now altered as requested and trust that version and OS build are orthogonal.

They should be (hopefully!) be orthogonal... In any case we'll need to drop support for Ubuntu 18.04 in the near future (it is reaching EOL) and we'll be back to only supporting a single OS when that happens.

Comment (mundaym) [reply doc/getting_started/_index.md:203]: @alees24 can you make this change? Once that is done I think we can merge this.

File: doc/getting_started/setup_verilator.md (hunk 3 above)
Comment (mundaym) [reply doc/getting_started/setup_verilator.md:33]: I think this is OK, it should always work on our supported system.

Comment (marnovandermaas) [reply doc/getting_started/setup_verilator.md:33]: Yeah, I saw that there is sudo in other places in the doc as well. So let's just keep it this way.

File: doc/getting_started/_index.md (hunk 4 above)
Comment (alees24) [reply doc/getting_started/_index.md]: Sorry, I did make that change at one point, but git and I got in a mess rebasing (my previous use of git has been simpler/more restricted) and the change must have been dropped in a reversion.

**Review by marnovandermaas — APPROVED** [review]
LGTM
